File: Algorithm/week5/Exam/B.py
n = int(input())
matrix = [list(map(int, input().split())) for _ in range(n)]

# Check diagonal elements
for i in range(n):
    if matrix[i][i] != 0:
        print("NO")
        exit()

# Check symmetry
for i in range(n):
    for j in range(i+1, n):
        if matrix[i][j] != matrix[j][i]:
            print("NO")
            exit()

# Check multiple edges
for i in range(n):
    for j in range(i+1, n):
        if matrix[i][j] > 1:
            print("NO")
            exit()

print("YES")

File: Algorithm/week5/Exam/B.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int N;
    cin >> N;

    vector<vector<int>> M(N, vector<int>(N));
    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j < N; j++)
        {
            cin >> M[i][j];
        }
    }

    // Check diagonal elements
    for (int i = 0; i < N; i++)
    {
        if (M[i][i] != 0)
        {
            cout << "NO" << endl;
            return 0;
        }
    }

    // Check symmetry
    for (int i = 0; i < N; i++)
    {
        for (int j = i + 1; j < N; j++)
        {
            if (M[i][j] != M[j][i])
            {
                cout << "NO" << endl;
                return 0;
            }
        }
    }

    // Check multiple edges
    for (int i = 0; i < N; i++)
    {
        for (int j = i + 1; j < N; j++)
        {
            if (M[i][j] > 1)
            {
                cout << "NO" << endl;
                return 0;
            }
        }
    }

    cout << "YES" << endl;
    return 0;
}
